pkg/fanal/analyzer/language/golang/mod/mod.go

```diff
@@ -164,8 +196,9 @@ func (a *gomodAnalyzer) fillAdditionalData(apps []types.Application) error {
 				apps[i].Packages[j].Licenses = licenseNames
 			}
 
-			// Collect dependencies of the direct dependency
-			if dep, err := a.collectDeps(modDir, lib.ID); err != nil {
+			// Collect dependencies of the direct dependency from $GOPATH/pkg/mod because the vendor directory doesn't have go.mod files.
+			gopathModDir := filepath.Join(gopath, "pkg", "mod", fmt.Sprintf("%s@%s", normalizeModName(lib.Name), lib.Version))
```

Comment:
We can add check here or in `collectDeps` func
```suggestion
			if !gopathModDirFound {
				return nil
			}
			// Collect dependencies of the direct dependency from $GOPATH/pkg/mod because the vendor directory doesn't have go.mod files.
			gopathModDir := filepath.Join(gopath, "pkg", "mod", fmt.Sprintf("%s@%s", normalizeModName(lib.Name), lib.Version))
```

Reply:
Thank you for the review. :)
I've addressed the other points and looked into this suggestion. 

It seems the check is already implemented here:
https://github.com/aquasecurity/trivy/blob/b7cbbdc0ce3f9bbdae999049af61281fe61edb06/pkg/fanal/analyzer/language/golang/mod/mod.go#L188-L198
Was there perhaps a different intention I might have missed?

Reply:
there are 2 point:
1. We try to use "early return" logic.
2. We will check `go.mod` file for each dependency (but you know that GOPATH dir doesn't exist), so this is extra resources.